core/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java

```diff
@@ -0,0 +1,111 @@
+                                  Map<String, ResponseCollectorService.ComputedNodeStats> nodeComputedStats) {
+        this.clientOutgoingConnections = clientConnections;
+        this.nodeComputedStats = nodeComputedStats;
+    }
+
+    public AdaptiveSelectionStats(StreamInput in) throws IOException {
+        this.clientOutgoingConnections = in.readMap(StreamInput::readString, StreamInput::readLong);
+        this.nodeComputedStats = in.readMap(StreamInput::readString, ResponseCollectorService.ComputedNodeStats::new);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeMap(this.clientOutgoingConnections, StreamOutput::writeString, StreamOutput::writeLong);
+        out.writeMap(this.nodeComputedStats, StreamOutput::writeString, (stream, stats) -> stats.writeTo(stream));
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject("adaptive_selection");
+        Set<String> allNodeIds = Sets.union(clientOutgoingConnections.keySet(), nodeComputedStats.keySet());
+        for (String nodeId : allNodeIds) {
+            builder.startObject(nodeId);
+            if (clientOutgoingConnections.containsKey(nodeId)) {
+            }
+            ResponseCollectorService.ComputedNodeStats stats = nodeComputedStats.get(nodeId);
+            if (stats != null) {
+                long outgoingSearches = clientOutgoingConnections.getOrDefault(nodeId, 0L);
+                builder.field("outgoing_searches", outgoingSearches);
+                builder.field("avg_queue_size", stats.queueSize);
+                builder.timeValueField("avg_service_time_ns", "avg_service_time", (long) stats.serviceTime, TimeUnit.NANOSECONDS);
+                builder.timeValueField("avg_response_time_ns", "avg_response_time", (long) stats.responseTime, TimeUnit.NANOSECONDS);
+                builder.field("rank", String.format(Locale.ROOT, "%.1f", stats.rank(outgoingSearches)));
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    /**
+     * Returns a map of node id to the outgoing search requests to that node
+     */
+    public Map<String, Long> getOutgoingConnections() {
+        return clientOutgoingConnections;
+    }
+
+    /**
+     * Returns a map of node id to the computed stats
+     */
+    public Map<String, ResponseCollectorService.ComputedNodeStats> getComputedStats() {
+        return nodeComputedStats;
+    }
+
+    /**
+     * Returns a map of node id to the ranking of the nodes based on the adaptive replica formula
+     */
+    public Map<String, Double> getRanks() {
+        Map<String, Double> ranks = new HashMap<>(nodeComputedStats.size());
+        nodeComputedStats.forEach((k, v) -> {
+            ranks.put(k, v.rank(clientOutgoingConnections.getOrDefault(k, 0L)));
+        });
+        return ranks;
```

Comment:
Is this clearer?

```diff
diff --git a/core/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java b/core/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java
index bed282b899..787fa1e9d1 100644
--- a/core/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java
+++ b/core/src/main/java/org/elasticsearch/node/AdaptiveSelectionStats.java
@@ -33,6 +33,7 @@ import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
 
 /**
  * Class representing statistics about adaptive replica selection. This includes
@@ -102,10 +103,12 @@ public class AdaptiveSelectionStats implements Writeable, ToXContentFragment {
      * Returns a map of node id to the ranking of the nodes based on the adaptive replica formula
      */
     public Map<String, Double> getRanks() {
-        Map<String, Double> ranks = new HashMap<>(nodeComputedStats.size());
-        nodeComputedStats.forEach((k, v) -> {
-            ranks.put(k, v.rank(clientOutgoingConnections.getOrDefault(k, 0L)));
-        });
-        return ranks;
+        return
+                nodeComputedStats
+                        .entrySet()
+                        .stream()
+                        .collect(Collectors.toMap(
+                                Map.Entry::getKey,
+                                e -> e.getValue().rank(clientOutgoingConnections.getOrDefault(e.getKey(), 0L))));
     }
 }
```

Reply:
Sure, thanks for the suggestion, I updated the method.